Testing/parsing/test_rate.py

```diff
@@ -0,0 +1,72 @@
+from eBCSgen.Parsing.ParseBCSL import Parser
+import Testing.parsing.objects_testing as objects
+from eBCSgen.Core.Complex import Complex
+
+parser = Parser("rate")
+
+
+def test_parser():
+    ret = parser.parse("1")
+    assert ret.success
+
+    ret = parser.parse("0.5")
+    assert ret.success
+
+    ret = parser.parse("2*3")
+    assert ret.success
+
+    ret = parser.parse("2**3")
+    assert ret.success
+
+    ret = parser.parse("2*[K()::cyt]")
+    assert ret.success
+
+    ret = parser.parse("[K()::cyt]**2")
+    assert ret.success
+
+    ret = parser.parse(
+        "(2 * [K()::cyt] + 3 * [B(T{s})::cell]) / (4 * [T{u}.T{_}::cell] - 2 * [B(T{s})::cyt])"
+    )
+    assert ret.success
+
+    # why is this expression incorrect, it passes separated into expr1 snd expr2
+    # ret = parser.parse(
+    #     "((([B(T{s})::cell] + [B(T{s})::cyt]) / ([B(T{_}.C())::cell] + [B()::cell])) / (([B(T{s}).T{s}::cyt] - [T{_}::cell]) / ([T{u}.T{_}::cell] - [B(T{s})::cell])))"
+    # )
+    expr1 = "([B(T{s})::cell] + [B(T{s})::cyt]) / ([B(T{_}).C()::cell] + [B()::cell])"
+    expr2 = (
+        "([B(T{s}).T{s}::cyt] - [T{_}::cell]) / ([T{u}.T{_}::cell] - [B(T{s})::cell])"
+    )
```

Comment:
Because currently only basic rational functions are allowed (i.e. where division is at most once). It is part of tasks in #70 to improve this.

Reply:
Understood, thanks for clarifying. I’ve updated the code and left a comment to note the current restriction on rational functions.